access/email/bot.go

```diff
@@ -0,0 +1,303 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gravitational/teleport-plugins/lib"
+	"github.com/gravitational/teleport-plugins/lib/logger"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"gopkg.in/mail.v2"
+
+	"github.com/mailgun/mailgun-go/v4"
+)
+
+var reviewReplyTemplate = template.Must(template.New("review reply").Parse(
+	`{{.Author}} reviewed the request at {{.Created.Format .TimeFormat}}.
+Resolution: {{.ProposedStateEmoji}} {{.ProposedState}}.
+{{if .Reason}}Reason: {{.Reason}}.{{end}}`,
+))
+
+// Bot is a email client that works with access.Request. Essentially it's not a bot, but let's
+// keep naming for the sake of uniformity.
+type Bot struct {
```

Comment:
Please don't call it `Bot`. It's unfortunate naming I'm trying to get rid of.

In pagerduty, jira and gitlab I already renamed it from `Bot` to `Client`.
In slack & mattermost it will probably remain a `Bot` but only because slack clients are actually bots :)
In your case I don't know what name is good, it's up to you to decide.

Reply:
Yes, you're absolutely correct, `Bot` sounds weird here. I just thought it's a convention. Will happily rename it to Client :-)